wallet/src/events.rs

```diff
@@ -192,12 +205,17 @@ pub fn record(kind: EventKind) {
     }
     EVENT_BUFFER.with(|buffer| {
         let mut buffer = buffer.borrow_mut();
-        let len = buffer.len();
+        let buffer = &mut *buffer;
+        let len = buffer.total();
         buffer.push(Event {
             id: len,
             timestamp: api::time() as u64,
             kind,
         });
+        if buffer.events.len() > 10_000 {
```

Comment:
Can we declare these constants somewhere (here, and the 1000 constant) instead of using values here?